api/pkg/s3/objectput.go

```diff
@@ -146,7 +148,14 @@ func (s *APIService) ObjectPut(request *restful.Request, response *restful.Respo
 			log.Debugln("finished read")
 			eof = true
 		}
-		err = stream.Send(&s3.PutObjectRequest{Data: buf[:n]})
+		// encrypt if needed
+		if bucketMeta.ServerSideEncryption.SseType == "SSE"{
+			key := bucketMeta.ServerSideEncryption.EncryptionKey
+			_, encbuf := utils.EncryptWithAES256RandomKey(buf, key)
```

Comment:
always with AES256RandomKey? No encryption config?

Reply:
this is the server managed key variant, so server picks a random key and uses AES-256, same as AWS model

Reply:
we will discuss. I meant whether we consider any other encryption or not.